include/envoy/server/options.h

```diff
@@ -12,6 +12,31 @@ namespace Envoy {
 namespace Server {
 
 /**
+ * Whether to run Envoy in serving mode, or in config validation mode at one of two levels (in which
+ * case we'll verify the configuration file is valid, print any errors, and exit without serving.)
+ */
+enum class Mode {
+  /**
+   * Default mode: Regular Envoy serving process. Configs are validated in the normal course of
+   * initialization, but if all is well we proceed to serve traffic.
+   */
+  Serve,
+
+  /**
+   * Validate as much as possible without opening network connections upstream or downstream.
+   */
+  Validate,
+
+  /**
+   * Mock out access to the filesystem. Perform no validation of files referenced in the config,
+   * such as runtime configs, SSL certs, etc. Validation will pass even if those files are malformed
+   * or don't exist, allowing the config to be validated in a non-prod environment.
+   * TODO(rlazarus): This isn't yet implemented.
```

Comment:
if it's not yet implemented, let's just delete the enum value, and leave the comment/TODO. We will need ValidateLight to replace the config_test stuff with this stuff.